lib/src/extensions/channel.dart

```diff
@@ -31,6 +37,9 @@ extension ThreadExtensions on Thread {
   /// {@macro paginated_endpoint_streaming_parameters}
   Stream<ThreadMember> streamThreadMembers({bool? withMembers, Snowflake? after, Snowflake? before, int? pageSize}) =>
       manager.streamThreadMembers(id, withMembers: withMembers, after: after, before: before, pageSize: pageSize);
+
+  /// Create a builder with the properties of this thread
+  ThreadBuilder toBuilder() => ThreadBuilder(name: name, type: type, autoArchiveDuration: autoArchiveDuration, rateLimitPerUser: rateLimitPerUser);
```

Comment:
Maybe we should also copy the `invitable` field if it's a private thread?

Reply:
I now check for the type of the Thread and append the invitable property to the constructor.